mailpoet/lib/EmailEditor/Engine/Renderer/Preprocessors/SpacingPreprocessor.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\EmailEditor\Engine\Renderer\Preprocessors;
+
+use MailPoet\EmailEditor\Engine\SettingsController;
+
+/**
+ * This preprocessor is responsible for setting default spacing values for blocks.
+ * In the early development phase, we are setting only margin-top for blocks that are not first or last in the columns block.
+ */
+class SpacingPreprocessor implements Preprocessor {
+  public function preprocess(array $parsedBlocks, array $layoutStyles): array {
+    $parsedBlocks = $this->addMarginTopToBlocks($parsedBlocks);
+    return $parsedBlocks;
+  }
+
+  private function addMarginTopToBlocks(array $parsedBlocks): array {
+    foreach ($parsedBlocks as $key => $block) {
+      // We don't want to add margin-top to column block
+      if ($key !== 0 && $block['blockName'] !== 'core/column') {
+        $block['email_attrs']['margin-top'] = SettingsController::FLEX_GAP;
+      }
```

Comment:
Why do we want to skip columns? Is it because of some specific behavior? Would it be possible to detect it differently than checking the block name? 

The thing is that here we are in the `Engine` namespace, so we should not use block names from the core integration in the code.

Maybe we could find some other common attributes that would help us detect when to skip adding the margin and it would work also for other blocks we will add later or alternatively we could use a filter (something like `$blocksWithoutMargin = apply_filters('mailpoet_xyz', [])`) and we could pass the columns there in `Integration/Core/Initializer`. What do you think?

Reply:
> Why do we want to skip columns? Is it because of some specific behavior? Would it be possible to detect it differently than checking the block name?

The main reason is that the `core/column` block is rendered horizontally compared to the rest of the blocks.
So the top margin could cause something like this:
![Screenshot 2024-01-23 at 17 18 44](https://github.com/mailpoet/mailpoet/assets/13644846/e9cc83d9-288c-402b-b6aa-7be09bc5f13f)

Nevertheless, I realized now the top margin is not used in the renderer for this block, so I think we can remove this part of the condition, and we don't need a filter here. Does it make sense to you?

Reply:
> Nevertheless, I realized now the top margin is not used in the renderer for this block, so I think we can remove this part of the condition, and we don't need a filter here. Does it make sense to you?

👍 If it is not needed, let's remove it. My concern was about mixing the engine and integration logic so if we remove the check for the specific block we are good.

Reply:
@costasovo I updated the condition in the last commit and rebased the branch due to conflicts with trunk.